### ISSUE TITLE
deprecation warning with Cython build_ext 

### ISSUE THREAD

**Issue by anthrotype** [pypa/setuptools#867]
When Cython is installed, the setuptools `build_ext` command tries to use `Cython.Distutils.build_ext` to build extension modules:

https://github.com/pypa/setuptools/blob/master/setuptools/command/build_ext.py#L17

With Cython 0.25, the distutils extension `Cython.Distutils.build_ext` was updated to use `cythonize` (which properly handles dependencies), and the old distutils extension deprecated and moved to `Cython.Distutils.old_build_ext`.

The change was temporarily reverted with Cython 0.25.1 to give projects time to migrate, but a deprecation warning is still printed when using Cython's `old_build_ext` as setuptools does.

```
...lib/python3.5/site-packages/Cython/Distutils/old_build_ext.py:30: UserWarning: Cython.Distutils.old_build_ext does not properly handle dependencies and is deprecated.
  "Cython.Distutils.old_build_ext does not properly handle dependencies "
```

I assume `setuptools.commands.build_ext` will have to be updated to import the `new_build_ext` from `Cython.Distutils.build_ext`.

**Comment by anthrotype** [comment]
Cf. Cython's changelog: https://github.com/cython/cython/blob/master/CHANGES.rst

**Comment by jaraco** [comment]
It's not clear to me from the changelog what is required of setuptools here. Try to use `new_build_ext` and then fall back to `build_ext` for older versions of Cython? Or just wait until future versions of Cython (including 0.25.0) expose `build_ext` as the preferred command? Can you put together a PR?

**Comment by ngoldbaum** [comment]
It looks like there's a hack in the cython code (that isn't working) that is supposed to suppress this error if the import is coming from setuptools:

https://github.com/cython/cython/blob/master/Cython/Distutils/old_build_ext.py#L36

And in fact, the code in cython's master branch is different now than it was in 0.25.1 as of 20 days ago:

https://github.com/cython/cython/commit/f5b89afa9d8ad1217640a40445fb88082cb40e3d

(possibly to fix this buggy behavior)

So I guess setuptools shouldn't do anything since this will be fixed in a future cython release. Note that I haven't actually verified if using an updated version of cython fixes this.

**Comment by jaraco** [comment]
Sounds good. I'll close as it sounds as if it will be fixed by Cython, but do re-open if there's something setuptools should do.

**Comment by ngoldbaum** [comment]
This is fixed in Cython 0.25.2, which was released this morning.

**Comment by anthrotype** [comment]
thanks
